api_app/serializers.py

```diff
@@ -77,18 +80,194 @@ class _AbstractJobCreateSerializer(rfs.ModelSerializer):
     connectors_requested = rfs.ListField(default=list)
     md5 = rfs.HiddenField(default=None)
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.warnings = []
+
     def validate(self, attrs: dict) -> dict:
         # check and validate runtime_configuration
         runtime_conf = attrs.get("runtime_configuration", {})
         if runtime_conf and isinstance(runtime_conf, list):
             runtime_conf = json.loads(runtime_conf[0])
         attrs["runtime_configuration"] = runtime_conf
+        return attrs
+
+    @staticmethod
+    def filter_analyzers(serialized_data: Dict, warnings: List) -> List[str]:
```

Comment:
Since you have decided to move this stuff, i would love that we remove the fact that we modify the parameters of the function, inside the function itself.
What i mean is pretty much change the signature to 
```python3
def filter_analyzers(serialized_data: Dict) -> Tuple[List[str], List[str]]:
```
And concatenate the `warnings` in the caller

Reply:
There are 2 approaches that can be taken here:
1) Return `warnings` from the filtering functions.
2) Set `self.warnings` in the filtering functions after making them non-static.
Which one would you prefer?
@0ssigeno @mlodic

Reply:
I'm ok with both approach!
Probably remove the @statichmethod is slightly better, but both works for me

Reply:
@0ssigeno @mlodic 
Alert: We can't normally use `self.warnings` here. This is because ListSerializer uses the same instance to serialize each of the entries. Hence, they all share a common `self.warnings`.

```
{
    "count": 2,
    "results": [
        {
            "job_id": 80,
            "status": "accepted",
            "warnings": [
                "Robtex_IP_Query won't be run because it does not support observable type domain."
            ],
            "analyzers_running": [
                "Classic_DNS",
                "Robtex_IP_Query"
            ],
            "connectors_running": []
        },
        {
            "job_id": 81,
            "status": "accepted",
            "warnings": [
                "Robtex_IP_Query won't be run because it does not support observable type domain."
            ],
            "analyzers_running": [
                "Classic_DNS"
            ],
            "connectors_running": []
        }
    ]
}
```

Reply:
Should I use method 1 now?

Reply:
Found a fix